pkg/kn/flags/resources_test.go

```diff
@@ -37,49 +37,75 @@ func parseQuantity(value string) resource.Quantity {
 
 func TestResourceOptions(t *testing.T) {
 	cases := []*resourceOptionsTestCase{
-		{"memory=200Mi,cpu=200m",
-			"memory=1024Mi,cpu=500m",
+		{[]string{"memory=200Mi", "cpu=200m"},
+			[]string{"memory=1024Mi", "cpu=500m"},
 			corev1.ResourceList{corev1.ResourceMemory: parseQuantity("200Mi"),
 				corev1.ResourceCPU: parseQuantity("200m")},
 			corev1.ResourceList{corev1.ResourceMemory: parseQuantity("1024Mi"),
 				corev1.ResourceCPU: parseQuantity("500m")},
 			false,
 		},
-		{"",
-			"nvidia.com/gpu=1",
+		{[]string{},
+			[]string{"nvidia.com/gpu=1"},
 			nil,
 			corev1.ResourceList{corev1.ResourceName("nvidia.com/gpu"): parseQuantity("1")},
 			false,
 		},
 
-		{"",
-			"memory:500Mi",
+		{[]string{},
+			[]string{"memory:500Mi"},
 			nil,
 			nil,
 			true,
 		},
-		{"memory:200Mi",
-			"",
+		{[]string{"memory:200Mi"},
+			[]string{},
 			nil,
 			nil,
 			true,
 		},
-		{"memory=200MB",
-			"",
+		{[]string{"memory=200MB"},
+			[]string{},
 			nil,
 			nil,
 			true,
 		},
+		{[]string{"cpu=500m", "cpu-"},
+			[]string{},
+			corev1.ResourceList{},
+			nil,
+			false,
+		},
+		{[]string{},
+			// resource being asked for removal is considered to be removed and its value isnt validated (200MB which is incorrect)
+			[]string{"memory=200Mi", "cpu=200m", "memory-=200MB"},
+			nil,
+			corev1.ResourceList{corev1.ResourceName("cpu"): parseQuantity("200m")},
+			false,
+		},
 	}
 	for _, c := range cases {
 		options := &ResourceOptions{}
 		options.Requests = c.requests
 		options.Limits = c.limits
-		err := options.Validate()
+		reqToRemove, limToRemove, err := options.Validate()
```

Comment:
Maybe don't call it validate as the main purpose is no to extract the values. What's about `options.Extract()` or `options.Parse()` (or `options.ParseAndValidate()`) ?

Reply:
Sure, will update the name when we unify the other similar flags.